src/clj/infinitelives/pixi/macros.clj

```diff
@@ -111,3 +112,33 @@
              (when ~form
                  ~@body
                  (recur ~test)))))
+
+(defmacro <!* [test & body]
+  `(let [result# (cljs.core.async/<! ~@body)]
+     (if ~test
+       result#
+       (throw (js/Error "go-while exit")))))
+
+(defmacro >!* [test & body]
+  `(let [result# (cljs.core.async/>! ~@body)]
+     (if ~test
+       result#
+       (throw (js/Error "go-while exit")))))
+
+(defn process-body [test [head & tail]]
+  (let [[processed-head processed-tail]
+          (cond
+            (identical? '() head) ['() tail]
+            (identical? '[] head) ['[] tail]
+            (list? head) [(process-body test head) tail]
+            (vector? head) [(vec (process-body test head)) tail]
+            (map? head) [(into {} (process-body test (seq head))) tail]
+            (= '<! head) ['baller.async/<!* (cons test tail)]
+            (= '>! head) ['baller.async/>!* (cons test tail)]
```

Comment:
Oops, I missed that. I recall noticing this earlier on but I must have forgotten